### PR TITLE
Upgrade ethers to v6

### PR DESCRIPTION
**Pull request by acolytec3**

### CODE REVIEW

File: packages/portalnetwork/src/client/provider.ts [hunk 1]
```diff
@@ -10,69 +10,68 @@ import {
...
   constructor(
-    fallbackProvider: string | ethers.providers.JsonRpcProvider,
+    fallbackProvider: string | ethers.JsonRpcProvider,
     network = 1,
     portal: PortalNetwork,
   ) {
     super(
-      typeof fallbackProvider === 'string' ? fallbackProvider : fallbackProvider.connection,
+      //@ts-ignore
+      typeof fallbackProvider === 'string' ? fallbackProvider : fallbackProvider.#connect.url,
```
Comment (ScottyPoi) [inline packages/portalnetwork/src/client/provider.ts:32]: what is going on with `fallbackProvider.#connect.url`?

Comment (ScottyPoi) [reply packages/portalnetwork/src/client/provider.ts:32]:
```suggestion
      typeof fallbackProvider === 'string' ? fallbackProvider : fallbackProvider.connection.url,
```

Comment (acolytec3) [reply packages/portalnetwork/src/client/provider.ts:32]: I dunno.  Still trying to work that out. The new and improved `ethers` is quite a bit different than v5.7 was.

**Review by ScottyPoi — APPROVED** [review]
LGTM!

**Pull request merged**